Message interviews:20 -  Read
From:      Paul Calder <calder@aramis.stanford.edu>
To:        <interviews@interviews.stanford.edu>
Reply-To:  <calder@lurch.stanford.edu>
Subject:   Problem stencilling bitmaps
Obsoletes: Jim Wight's message of Mon, 3 Jul 89 12:17:53 +0100

Jim has found a bug in Painter::Stencil...

When a mask bitmap was specified in the X11 Painter::Stencil function
(and the mask was different from the data bitmap), the mask was
ignored because the wrong graphics context was used.
--

Apply with ``cd iv; patch -p -N < diffs'' to patch files:

        src/libInterViews/X11.c

*** /tmp/,RCSt1a00788             Tue Jul  4 12:13:46 1989
--- src/libInterViews/X11.c       Tue Jul  4 12:13:23 1989
***************
*** 1345,1351 ****
              XSetClipMask(_world->display(), gc, (Pixmap)m->Map());
              XCopyPlane(
                  _world->display(), (Pixmap)d->Map(), (Drawable)c->Id(),
!                 rep->fillgc, 0, 0, d->Width(), d->Height(), dx, dy, 1
              );
              XFreeGC(_world->display(), gc);
          }
--- 1345,1351 ----
              XSetClipMask(_world->display(), gc, (Pixmap)m->Map());
              XCopyPlane(
                  _world->display(), (Pixmap)d->Map(), (Drawable)c->Id(),
!                 gc, 0, 0, d->Width(), d->Height(), dx, dy, 1
              );
              XFreeGC(_world->display(), gc);
          }
